### PR TITLE
Fix status parser for direct streaming mode

### PR DESCRIPTION
**Pull request by aromanielloNTIA**
Problem: in direct IQ streaming mode (using `IQSTREAM_Acquire`), the returned status code does not conform to the Tektronix API documentation. As of v1.3.0 of this API wrapper, this results in "No error" status being returned when ADC overrange events occur. The problem does not affect `IQSTREAM_Tempfile` methods in v1.3.0. It is provable that returned codes do not conform to the documentation since non-zero bits occur in places 6-15 and 22-31, which are unused and should always be zero according to the programming manual. It could be possible that the ctypes structure being used to retrieve these results is incorrectly configured; if this is the case, it is because neither the programming manual nor example code provided by Tektronix implement this structure correctly. Without being able to see the source code of the API, this problem remains unfixable at its core until Tektronix issues an update to the RSA API for Linux.

Solution: This PR updates the status parser so that ADC overrange events are always correctly flagged. This has been confirmed through testing with both `IQSTREAM_Acquire` and `IQSTREAM_Tempfile` methods on RSA306B and RSA507A hardware. The "USB data stream discontinuity" status flag is erroneously returned by streaming IQ direct to client, so it has been removed from the status parser. Additionally, status codes are sometimes returned which do not match any codes recorded in the documentation. A catchall status response has been added to handle these. It is recommended that applications using this API wrapper take care in their handling of return statuses. Return statuses for no error and overrange appear to always be correct, however it seems that unknown codes are sometimes returned when there is no error as well. Use of the Tempfile methods seems not to yield unknown status codes, and is a viable way to circumvent this problem

For more context see the parser example offered by Tektronix [here](https://github.com/tektronix/RSA_API/blob/712cf5c84b22b2d8095984414bffc423be06bd19/Python/rsa_api_full_example.py#L411)

### CODE REVIEW

**Comment by aromanielloNTIA** [comment]
Tektronix has archived their RSA API [repository](https://github.com/tektronix/RSA_API) as of yesterday. I have reached out to Tektronix to ask whether development is officially ceased. In any case I do not expect the underlying issues here to be addressed in the near future.

File: src/rsa_api/rsa_api.py [hunk 1]
```diff
@@ -2406,6 +2412,8 @@ def IQSTREAM_StatusParser(
             if bool(status & 0x200000):  # mask bit 21
                 status_str += "Output buffer overflow. File writing too slow, "
                 status_str += "data loss has occurred.\n"
+            if status_str == "":
```
Comment (jhazentia) [inline src/rsa_api/rsa_api.py:2415]: It seems like if there is a known error and unknown error, only the known error will be reported. Should it be changed such that unknown error will also be reported?

Comment (aromanielloNTIA) [reply src/rsa_api/rsa_api.py]: This is now fixed

**Comment by aromanielloNTIA** [comment]
Re-running CI checks:

pre-commit.ci run

File: src/rsa_api/rsa_api.py [hunk 2]
```diff
@@ -2406,12 +2406,95 @@ def IQSTREAM_StatusParser(
...
+          too slow. Data loss has occurred.
+        - Invalid status code returned. Some always-zero bits are nonzero.
+
+        In the case of multiple status codes being returned, the status
+        string will contain all returned status strings, separated by line
+        breaks.
+
+        Parameters
+        ----------
+        iq_stream_info : _IQStreamFileInfo
```
Comment (jhazentia) [inline src/rsa_api/rsa_api.py:2446]: This should be _IQStreamIQInfo

Comment (aromanielloNTIA) [reply src/rsa_api/rsa_api.py]: Fixed in [4f31eb9](https://github.com/NTIA/tekrsa-api-wrap/pull/14/commits/4f31eb905604937550d5a4b3b726459a4568eded)

**Review by jhazentia — APPROVED** [review]

**Pull request merged**